src/Stratis.Bitcoin/P2P/Peer/NetworkPeer.cs

```diff
@@ -587,14 +590,14 @@ private void CallMessageReceivedHandlers(IncomingMessage message)
                 this.logger.LogDebug("Connection to self detected and will be aborted.");
 
                 VersionPayload versionPayload = this.Parameters.CreateVersion(this.PeerAddress.Endpoint, this.Network, this.dateTimeProvider.GetTimeOffset());
-                await this.SendMessageAsync(versionPayload);
+                await this.SendMessageAsync(versionPayload, cancellation).ConfigureAwait(false);
                 this.Disconnect("Connected to self");
 
                 this.logger.LogTrace("(-)[CONNECTED_TO_SELF]");
                 return;
             }
 
-            using (CancellationTokenSource cancellationSource = CancellationTokenSource.CreateLinkedTokenSource(this.Connection.CancellationSource.Token))
+            using (CancellationTokenSource cancellationSource = CancellationTokenSource.CreateLinkedTokenSource(this.Connection.CancellationSource.Token, cancellation))
             {
                 cancellationSource.CancelAfter(TimeSpan.FromSeconds(10.0));
```

Comment:
`this.cancellationSource.Cancel...`

Reply:
i can't see such a conflicting field, where is it?

Reply:
oh, sry, looks like this is smth that was removed already too.